tcping.go

```diff
@@ -417,17 +416,19 @@ func newNetworkInterface(tcpStats *stats, netInterface string) networkInterface
 		// Iterating through the available addresses to identify valid IP configurations
 		for _, addr := range addrs {
 			if ip := addr.(*net.IPNet).IP; ip != nil {
-				// default is ipv4
-				if ip.To4() != nil && !tcpStats.userInput.useIPv6 {
-					// ipv4
+				// netip.Addr
+				nipAddr, err := netip.ParseAddr(ip.String())
+				if err != nil {
+					continue
+				}
+
+				if nipAddr.Is4() && !tcpStats.userInput.useIPv6 {
 					interfaceAddress = ip
 					break
-				} else if ip.To4() == nil && tcpStats.userInput.useIPv6 {
-					// ipv6
+				} else if nipAddr.Is6() && tcpStats.userInput.useIPv6 {
```

Comment:
```suggestion
				} else if nipAddr.Is6() && !tcpStats.userInput.useIPv4 {
```

This way some corner cases get covered